firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistence.java

```diff
@@ -151,22 +178,55 @@ public void finalizeReports(String currentSessionId) {
   }
 
   public List<CrashlyticsReport> loadFinalizedReports() {
-    final List<CrashlyticsReport> reports = new ArrayList<>();
-    final List<File> fatalReports = getAllFilesInDirectory(fatalReportsDirectory);
-    for (File reportFile : fatalReports) {
-      reports.add(TRANSFORM.reportFromJson(readTextFile(reportFile)));
+    final List<CrashlyticsReport> allReports = new ArrayList<>();
+    final List<File> priorityReports = getAllFilesInDirectory(priorityReportsDirectory);
+    for (File reportFile : priorityReports) {
+      allReports.add(TRANSFORM.reportFromJson(readTextFile(reportFile)));
     }
-    final List<File> nonFatalReports = getAllFilesInDirectory(nonFatalReportsDirectory);
-    for (File reportFile : nonFatalReports) {
-      reports.add(TRANSFORM.reportFromJson(readTextFile(reportFile)));
+    final List<File> reports = getAllFilesInDirectory(reportsDirectory);
+    for (File reportFile : reports) {
+      allReports.add(TRANSFORM.reportFromJson(readTextFile(reportFile)));
     }
-    return reports;
+    return allReports;
+  }
+
+  private static boolean isHighPriorityEvent(String eventFileName) {
+    return eventFileName.endsWith(PRIORITY_EVENT_SUFFIX);
+  }
+
+  private String generateEventFilename(boolean isHighPriority) {
+    final String eventNumber =
+        String.format(Locale.US, EVENT_COUNTER_FORMAT, eventCounter.getAndIncrement());
+    final String prioritySuffix = isHighPriority ? PRIORITY_EVENT_SUFFIX : NORMAL_EVENT_SUFFIX;
+    return EVENT_FILE_NAME_PREFIX + eventNumber + prioritySuffix;
   }
 
   private File getSessionDirectoryById(String sessionId) {
     return new File(openSessionsDirectory, sessionId);
   }
 
+  private static int trimEvents(File sessionDirectory, int maximum) {
+    final List<File> normalPriorityEventFiles =
+        getFilesInDirectory(
+            sessionDirectory, CrashlyticsReportPersistence::isNormalPriorityEventFile);
+    Collections.sort(normalPriorityEventFiles, CrashlyticsReportPersistence::oldestEventFileFirst);
```

Comment:
getFilesInDirectory makes an array internally, did we consider just working with the array class? (e.g., by doing Arrays.sort)

Reply:
I think, in general, working with lists is simpler than working with arrays, and in this case I'm not sure there's much performance benefit either way. Happy to discuss it further if you feel strongly about it, though.